django/cantusdb_project/main_app/views/views.py

```diff
@@ -620,3 +625,40 @@ def content_overview(request):
     return render(
         request, "content_overview.html", {"objects": recently_updated_50_objects}
     )
+
+def redirect_node_url(request, pk):
+    not_found = handle404(request, HttpResponseNotFound)
+    if pk >= 1_000_000:
+        return not_found
+    
+    # chant, source, sequence, article
+    possible_types = [
+        (Chant, 'chant-detail'),
+        (Source, 'source-detail'),
+        (Sequence, 'sequence-detail'),
+        (Article, 'article-detail')
+    ]
+
+    user_id = get_user_id(pk)
+    if get_user_id(pk) is not None:
+        return redirect('user-detail', user_id)
+    
+    response = next((redirect(view, pk) for (rec_type, view) in possible_types if record_exists(rec_type, pk)), not_found)
+    return response
+
+# used to determine whether record of specific type (chant, source, sequence, article) exists for a given pk
+def record_exists(rec_type, pk):
+    try:
+        result = rec_type.objects.get(id=pk)
+        print(f'result: {result}')
+        return True
+    except rec_type.DoesNotExist:
+        return False
+
+# special case for User (using indexer ID)
```

Comment:
can this comment be fleshed out a bit, and maybe turned from a comment into a docstring? Something along the lines of "indexers were originally stored in the general Node table in OldCantus, but now they're represented as users, so we need to check `old_indexer_id` rather than `id`"?